tests/test_data_loader.py

```diff
@@ -215,5 +236,72 @@ def get_data_source(*args, **kwargs):
         kpi_info, end_date=end_date, start_date=start_date
     )
     end_date = end_date + timedelta(days=1)
-    output_query = f"""select * from `cloud_cost` where `date` >= '{start_date.strftime("%Y-%m-%d")}T00:00:00+05:30' and `date` < '{end_date.strftime("%Y-%m-%d")}T00:00:00+05:30'"""
+    output_query = """select * from `cloud_cost` where `date` >= """ \
+        + f"""'{start_date.strftime("%Y-%m-%d")}T00:00:00+05:30' and `date` < """ \
+        + f"""'{end_date.strftime("%Y-%m-%d")}T00:00:00+05:30'"""
+    assert output_query == dl._build_query().strip()
+
+    # test for snowflake
```

Comment:
Some feedback for the future (can be added as TODO)
- Most of this code feels repetitive. This can be made a generic function and then a list of test case input and output can be passed to have the test cases. In that way, you can easily add more than one test case for a single data source.
- We can add 3-4 more test cases for the snowflake to be sure that we have covered the majority of test cases.